internal/builder/builder_test.go

```diff
@@ -920,8 +920,37 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 					h.AssertTrue(t, strings.Contains(layers[2], h.LayerFileName(ext2v1)))
 				})
 			})
-		})
 
+			when("system buildpacks", func() {
+				it.Before(func() {
+					subject.SetLifecycle(mockLifecycle)
+					subject.AddBuildpack(bp1v1)
+					subject.SetSystem(dist.System{
+						Pre: dist.SystemBuildpacks{
+							Buildpacks: []dist.ModuleRef{
+								{ModuleInfo: dist.ModuleInfo{ID: bp1v1.Descriptor().Info().ID}}},
+						},
+					})
+				})
+
+				it("should write system buildpacks to system.toml)", func() {
+					err := subject.Save(logger, builder.CreatorMetadata{})
+					h.AssertNil(t, err)
+
+					layerTar, err := baseImage.FindLayerWithPath("/cnb/system.toml")
+					h.AssertNil(t, err)
+					h.AssertOnTarEntry(t, layerTar, "/cnb/system.toml", h.ContentEquals(`[system]
+  [system.pre]
```

Comment:
@natalieparellano could you give me a hint here, I don't know how to avoid writing this key to the `system.toml` . Did I do something wrong in the [dist](https://github.com/buildpacks/pack/pull/2349/files#diff-8fe04731e3b84061e2db8576200449a5aabc8f4aaf1957c6ce50f0231295f22a) file?

Reply:
Not sure I follow - looks like valid TOML to me. Are you trying to avoid specifying the super tables? These are probably being inserted by our serializer.

Reply:
When I write the system.toml I ended up with
```toml
[system]
  [system.pre]
    [[system.pre.buildpacks]]
        id = "buildpack-1-id"
        version = "buildpack-1-version-1"
```

but we probably don't want the `system.pre` to be there, just:
```toml
[system]
    [[system.pre.buildpacks]]
        id = "buildpack-1-id"
        version = "buildpack-1-version-1"
```
if it is not gonna hurt anybody, I am ok with living it as it is, but I wasn't sure

Reply:
@jkutner, I am trying to implement the System Buildpacks [RFC](https://github.com/buildpacks/rfcs/blob/main/text/0101-system-buildpacks-in-builder.md#systemtoml-toml) and I have some doubts with the `system.toml`